src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -779,6 +781,33 @@ public void addAvatar(Avatar avatar) {
         addAvatar(avatar, true);
     }
 
+    public boolean changeAvatarElement(int elementTypeId){
+        EntityAvatar mainCharacterEntity = getTeamManager().getCurrentAvatarEntity();
+        Avatar mainCharacter = mainCharacterEntity.getAvatar();
+        // if no candidate skill to change
+        if (mainCharacter.getData().getCandSkillDepotIds() == null){
```

Comment:
Might be good to also check the bounds here, to prevent an out of bounds read on the array with a wrong value.